iron-list.html

```diff
@@ -777,6 +831,25 @@
     },
 
     /**
+     * This function will return the amount the physical
+     * top should be incremented based on it's current iterative
+     * state in a given row. This function will only return a 
+     * non-zero value when the index modulous `_physicalItemsPerRow` 
+     * is one less than the possible `_physicalItemsPerRow`
+     */
+    _getPhysicalTopIncrement: function(kth) {
+      if (!Array.isArray(this.items)) return 0;
+      if (kth % this._physicalItemsPerRow !== (this._physicalItemsPerRow - 1)) return 0;
+      var rowSize = 0;
+      for (length = kth + this._physicalItemsPerRow,
+           maxLength = this.items.length;
+           kth < maxLength && kth < length; kth++) {
+        rowSize = this._physicalSizes[kth] > rowSize ? this._physicalSizes[kth] : rowSize;
+      }

```

Comment:
Hmm. Giving a row, how would it estimate the top position for the items that have different heights? If they have different height, I think the user would want to vertically align the items in a row. Hence, this question: should we only accept fixed height for the `grid` case? 
e.g. 
```js
this._itemHeight =  this._physicalCount > 0 ? this._physicalItems[0].offsetHeight :  
    SOME_HEIGHT_DEFAULT_VALUE;
```